### DISCUSSION TITLE
How to work with gd.ArrayOf Count method?

### DISCUSSION THREAD

**Discussion by ullanar** [quaadgras/graphics.gd#52] [Q&A] [answered]
Hi! Also question about `gd.ArrayOf`.`Count` method

Firstly I thought what it should be analog of `len()` func. But it requires some `value` of same type as elements. And I stopped to understand what is going on x)

Can u describe please how to work with it?

```go
func (gd.ArrayOf[gd.Area2D]) Count(value gd.Area2D) int64
```

Also, dont you think to create func which makes go array from gd.Array? Maybe it is my personal feeling but I feel like it would be much easier to work with native go array which will allow us to do all kinds of classic go crazy things

**Comment by Splizard** [comment, marked as answer]
The count method appears to return the number of elements that match the given value that you provide 
https://docs.godotengine.org/en/stable/classes/class_array.html#class-array-method-count

You can think of it being implemented like this:
```go 
type array[T comparable] []T

func (a array[T]) count(value T) (n int) {
     for _, elem := range a {
         if elem == value { 
               n++
         }
     }
     return n
}
```

You can copy a Godot array into a Go array with this function. I will consider adding this to `gd` but it is likely a bit slower than using the Godot array directly (but as you mention, it will be easier to work with).

```go
func ToGoSlice[T any](lt gd.Lifetime, array gd.ArrayOf[T]) []T {
    var slice = make([]T, array.Size())
    for i := range array.Size() {
        slice[i] = array.Index(lt, gd.Int(i))
    }
    return slice
}
```

**Reply by ullanar** [reply]
Thanks! About `ToGoSlice` - I do not see `.Len()` method in `gd.ArrayOf` type

**Reply by Splizard** [reply]
Ah yes, Godot calls this `Size` I believe, have updated my answer.
